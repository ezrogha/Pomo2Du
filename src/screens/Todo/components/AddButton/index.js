import React from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export default ({ navigate }) => (
    <TouchableOpacity style={styles.iconContainer} onPress={navigate(this, 'TodoDetails')}>
        <Ionicons name="ios-add" size={36} color="white" />
    </TouchableOpacity>
)

const styles = StyleSheet.create({
    iconContainer: {
        marginLeft: 20,
        marginBottom: 5,
        padding: 5,
        shadowColor: "transparent"
    }
})
