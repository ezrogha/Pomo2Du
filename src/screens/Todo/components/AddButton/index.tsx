import React from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

interface Props {
    navigate: (screen: string) => void
}

export default ({ navigate }: Props) => (
    <TouchableOpacity style={styles.iconContainer} onPress={() => navigate('TodoDetails')}>
        <Ionicons name="ios-add" size={36} color="white" />
    </TouchableOpacity>
)

const styles = StyleSheet.create({
    iconContainer: {
        marginLeft: 20,
        marginBottom: 5,
        padding: 5,
        shadowColor: "transparent"
    }
})
