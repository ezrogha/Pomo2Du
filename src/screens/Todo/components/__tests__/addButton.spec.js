import React from 'react'

import AddButton from '../AddButton';
import renderer from 'react-test-renderer'

describe("Testing AddButton Component", () => {
    test("snapshot test", () => {
        const props = {
            navigate: () => {}
        }
        const tree = renderer.create(<AddButton { ...props }/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
