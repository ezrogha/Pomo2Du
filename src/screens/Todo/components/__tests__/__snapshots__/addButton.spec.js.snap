// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddButton Component snapshot test 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginBottom": 5,
      "marginLeft": 20,
      "opacity": 1,
      "padding": 5,
      "shadowColor": "transparent",
    }
  }
>
  <Text />
</View>
`;
